src/handlers/transfers/handler.js

```diff
@@ -726,7 +730,7 @@ const processFxFulfilMessage = async (message, functionality, span) => {
     ['success', 'funcName']
   ).startTimer()
 
-  const dupCheckResult = await fxFulfilService.getDuplicateCheckResult({ commitRequestId, payload })
+  const dupCheckResult = await fxFulfilService.getDuplicateCheckResult({ commitRequestId, payload, action })
```

Comment:
This is the actual fix. `fxFulfilService.getDuplicateCheckResult` already implements correct duplicate error check based on `action`. Previously, `action` was not passed in.